app/assets/javascripts/application.js

```diff
@@ -667,23 +667,32 @@ function renderMorpheus(dataPath, annotPath, selectedAnnot, selectedAnnotType, t
 }
 
 // toggles visibility and disabled status of file upload and fastq url fields
-function toggleFastqFields(target) {
+function toggleFastqFields(target, state) {
     var selector = $("#" + target);
     var fileField = selector.find('.upload-field');
-    $(fileField).toggleClass('hidden');
     var fastqField = selector.find('.fastq-field');
-    $(fastqField).toggleClass('hidden');
-    // toggle disabled status by returning inverse of current state
-    $(fastqField).find('input').attr('disabled', !$(fastqField).find('input').is('[disabled=disabled]'));
-    // set human data attr to true
     var humanData = $(fastqField).find('input[type=hidden]');
-    $(humanData).val($(humanData).val() === 'true' ? 'false' : 'true' );
-    // enable name field & update button to allow saving
     var saveBtn = selector.find('.save-study-file');
-    $(saveBtn).attr('disabled', !$(saveBtn).is('[disabled=disabled]'));
     var nameField = selector.find('.filename');
```

Comment:
It looks like `nameField` (and `fastqField`, `humanData`, `fileField` and `saveBtn`) are only used as jQuery objects.  If that's true, then perhaps consider getting each's jQuery object as part of the initial assignment.

E.g.:
`var nameField = $(selector.find('.filename'));`

That would enable cleaner references to those variables later, e.g. `nameField.attr('readonly', false);` instead of `$(nameField).attr('readonly', false);`.

Not a blocking issue; just a possible minor readability enhancement.

Reply:
Sounds reasonable, and it does cut down on the proliferation of $()